contracts/src/v0.1/L1Endpoint.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./VRFConsumerBase.sol";
+import "./interfaces/IVRFCoordinator.sol";
+
+contract L1Endpoint is Ownable, VRFConsumerBase {
+    IVRFCoordinator COORDINATOR;
+    mapping(address => uint256) private _balance;
+    mapping(address => bool) private _oracles;
+    mapping(uint256 => address) private _requestIdRequester;
+    uint256 private _fee;
+
+    error InsufficientBalance();
+    error OnlyOracle();
+
+    event FeeUpdated(uint256 oldFee, uint256 newFee);
+    event OracleAdded(address oracle);
+    event OracleRemoved(address oracle);
+
+    event RandomWordRequested(uint256 requestId, address requester);
+    event RandomWordFulfilled(uint256 requestId, address requester, uint256[] randomWords);
+
+    constructor(address coordinator) VRFConsumerBase(coordinator) {
+        COORDINATOR = IVRFCoordinator(coordinator);
+    }
+
+    receive() external payable {
+        _balance[msg.sender] += msg.value;
```

Comment:
I am not sure about exact logic of payment. However how about include withdraw function for the users?

Reply:
I thought about it, but then decided we should not let them for now. 
In that case, we need to track pending requests (can not let them withdraw if they have a pending request). This will bring a lot of headache since the request may stuck on L1, L2 or somewhere in the middle.
It's a nice to have feature, and maybe we'll revive this after having the MVP up and running

Reply:
Thanks. That is good points. 
Let's create new issue or put TODO task in code, then we can work on it.